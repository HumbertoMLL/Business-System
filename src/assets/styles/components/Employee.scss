@import '../Vars.scss';

.section__list-users--component {
  background-color: $color-suave;
  width: 100%;
  height: 100%;
  overflow: auto;

&-data {
  background-color: $color-suave;
  border-radius: 8px;
  list-style: none;
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  margin-right: 4em;
  margin-left: .5em;
  padding-right: .5em;

& li {
  margin-left: 4em;
  margin-right: .5em;

}
&-img {
& img{
  width: 5.2em;
  height: 5.2em;
}
}
&-employe {
  & h2 {
  font-family: $fuente;
  font-size: 20px;
  }
  & h3 {
    font-family: $fuente;
    font-size: 16px;
  }

}
&-salary {
  & h2 {
    font-family: $fuente;
    font-size: 20px;
    }
    & h3 {
      font-family: $fuente;
      font-size: 16px;
    }

}
&-status {
  font-size: 20px;
  font-family: $fuente;

}
&-edit {
  display: flex;
  flex-direction: row;

  button {
    margin-left: 1em;
    background-color: $color-suave;
    border: none;

    &:focus {
      outline: none;
    }
  }
& img {
  width: 2em;
  height: auto;
  margin-left: .5em;
}
      }
    }
}
