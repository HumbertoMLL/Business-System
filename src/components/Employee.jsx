import React from 'react';
import PropTypes from 'prop-types';
import '../assets/styles/components/Employee.scss';
// import pencil from '../static/pencil.svg';
// import trash from '../static/trash.svg';
// import userIcon from '../static/woman.svg';

const Employee = ({ name, proyect, salary, monthsOfWork }) => (
  <span className='section__list-users--component'>
    <ul className='section__list-users--component-data'>
      <li className='section__list-users--component-data-img'>
        <img src='https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/woman.svg?alt=media&token=28e58c45-62a1-4c30-b2a3-da9dac2d692f' alt='User_image' />
      </li>
      <li className='section__list-users--component-data-employe'>
        <h2>{ name }</h2>
        <h3> Software Engineer </h3>
      </li>
      <li className='section__list-users--component-data-salary'>
        <h2>{salary}</h2>
        <h3>{proyect}</h3>
      </li>
      <li className='section__list-users--component-data-status'>
        {monthsOfWork}
      </li>
      <li className='section__list-users--component-data-edit'>
        <button type='button'>
          <img src='https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/pencil.svg?alt=media&token=75fd35e8-495d-4ce6-9f75-f32a36cc1fcd' alt='Edit Icon' />
        </button>
        <button type='button'>
          <img src='https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/trash.svg?alt=media&token=0d246700-9a88-4fc2-b267-649d7be7a9fd' alt='Delete Icon' />
        </button>
      </li>
    </ul>
  </span>
);

Employee.propTypes = {
  name: PropTypes.string,
  proyect: PropTypes.string,
  salary: PropTypes.number,
  monthsOfWork: PropTypes.number,
};

export default Employee;
