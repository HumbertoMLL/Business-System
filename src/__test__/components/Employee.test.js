import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Employee from '../../components/Employee';

describe('<Employee />', () => {
  test('Render del componente Employee', () => {
    const employee = mount(<Employee />);
    expect(employee.length).toEqual(1);
  });
});

describe('Employee Snapshot', () => {
  test('Comprobar la UI del componente Employee', () => {
    const employee = create(<Employee />);
    expect(employee.toJSON()).toMatchSnapshot();
  });
});
//Para actualizar el snapshot cuando actualizamos el html
//Usamos el comando en la terminal
//jest --updateSnapshot
