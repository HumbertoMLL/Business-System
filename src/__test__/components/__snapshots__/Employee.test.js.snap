// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Snapshot Comprobar la UI del componente Employee 1`] = `
<span
  className="section__list-users--component"
>
  <ul
    className="section__list-users--component-data"
  >
    <li
      className="section__list-users--component-data-img"
    >
      <img
        alt="User_image"
        src="https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/woman.svg?alt=media&token=28e58c45-62a1-4c30-b2a3-da9dac2d692f"
      />
    </li>
    <li
      className="section__list-users--component-data-employe"
    >
      <h2 />
      <h3>
         Software Engineer 
      </h3>
    </li>
    <li
      className="section__list-users--component-data-salary"
    >
      <h2 />
      <h3 />
    </li>
    <li
      className="section__list-users--component-data-status"
    />
    <li
      className="section__list-users--component-data-edit"
    >
      <button
        type="button"
      >
        <img
          alt="Edit Icon"
          src="https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/pencil.svg?alt=media&token=75fd35e8-495d-4ce6-9f75-f32a36cc1fcd"
        />
      </button>
      <button
        type="button"
      >
        <img
          alt="Delete Icon"
          src="https://firebasestorage.googleapis.com/v0/b/humbertostark21.appspot.com/o/trash.svg?alt=media&token=0d246700-9a88-4fc2-b267-649d7be7a9fd"
        />
      </button>
    </li>
  </ul>
</span>
`;
